pandas/core/groupby.py

```diff
@@ -380,6 +384,38 @@ def indices(self):
         self._assure_grouper()
         return self.grouper.indices
 
+    @property
```

Comment:
I understand @jreback 's point is something like below. ``setter`` and assignments are not needed.

```
@property 
def inputconstructor(self):
    return self.obj._constructor
```

In addition, my opinion is property is not very beneficial because it doesn't simplify variable names. @jreback?